src/createStore.js

```diff
@@ -101,6 +102,15 @@ export default function createStore(reducer, initialState) {
     try {
       isDispatching = true;
       currentState = currentReducer(currentState, action);
+
+      if(currentState instanceof StateAndEffect) {
+        var effect = currentState.effect;
+        // Since side effects may dispatch at any time, don't run them
+        // immediately since we don't want cascading dispatches
+        setTimeout(() => effect(dispatch, getState), 0);
```

Comment:
This will run all the effects produced in a single dispatch cycle on the next tick (the effects are combined into a single effect by `combineReducers`). I think that should be ok. They can't really rely on each other, so they aren't racey. They will only just fire off async stuff.

Not sure how I feel about event delaying this a tick though, because async calls will be started a tick later, but I guess async code isn't that sensitive to perf. I don't see how we could get around delaying this because `dispatch` could be called immediately (is that a bad thing?)